pkl-core/src/main/java/org/pkl/core/EvaluatorBuilder.java

```diff
@@ -129,301 +133,695 @@ private static void addSystemProperties(EvaluatorBuilder builder) {
+  /**
+   * Returns the logger for trace and warn messages.
+   *
+   * <p>For more information, see {@link #setLogger}.
+   */
   public Logger getLogger() {
     return logger;
   }
 
   /**
-   * Adds the given module key factory. Factories will be asked to resolve module keys in the order
-   * they have been added to this builder.
+   * Adds a {@code ModuleKeyFactory} object that handles module imports.
+   *
+   * <p>For more information, see {@link #setModuleKeyFactories}.
+   *
+   * @see #addModuleKeyFactories
+   * @see #getModuleKeyFactories
    */
   public EvaluatorBuilder addModuleKeyFactory(ModuleKeyFactory factory) {
     moduleKeyFactories.add(factory);
     return this;
   }
 
   /**
-   * Adds the given module key factories. Factories will be asked to resolve module keys in the
-   * order they have been added to this builder.
+   * Adds {@code ModuleKeyFactory} objects that handle module imports.
+   *
+   * <p>For more information, see {@link #setModuleKeyFactories}.
+   *
+   * @see #addModuleKeyFactory
+   * @see #getModuleKeyFactories
    */
   public EvaluatorBuilder addModuleKeyFactories(Collection<ModuleKeyFactory> factories) {
     moduleKeyFactories.addAll(factories);
     return this;
   }
 
-  /** Removes any existing module key factories, then adds the given factories. */
+  /**
+   * Sets the {@code ModuleKeyFactory} objects that handle module imports.
+   *
+   * <p>Pkl code can import a module with {@code import "moduleUri"}, {@code amends "moduleUri},
+   * {@code extends "moduleUri"}, or an {@code import("moduleUri")} expression. The import is
+   * handled by the first element in {@code factories} (according to iteration order) that can
+   * handle {@code moduleUri}.
+   *
+   * <p>Default when {@link #unconfigured}: empty list
+   *
+   * <p>Default when {@link #preconfigured}:
+   *
+   * <ul>
+   *   <li>{@link ModuleKeyFactories#standardLibrary}
+   *   <li>{@link ModuleKeyFactories#classPath}
+   *   <li>{@link ModuleKeyFactories#fromServiceProviders}
+   *   <li>{@link ModuleKeyFactories#file}
+   *   <li>{@link ModuleKeyFactories#pkg}
+   *   <li>{@link ModuleKeyFactories#projectpackage}
+   *   <li>{@link ModuleKeyFactories#genericUrl}
+   * </ul>
+   *
+   * @see #addModuleKeyFactory
+   * @see #addModuleKeyFactories
+   * @see #getModuleKeyFactories
+   */
   public EvaluatorBuilder setModuleKeyFactories(Collection<ModuleKeyFactory> factories) {
     moduleKeyFactories.clear();
     return addModuleKeyFactories(factories);
   }
 
-  /** Returns the currently set module key factories. */
+  /**
+   * Returns the {@code ModuleKeyFactory} objects that handle module imports.
+   *
+   * <p>For more information, see {@link #setModuleKeyFactories}.
+   *
+   * @see #addModuleKeyFactory
+   * @see #addModuleKeyFactories
+   */
   public List<ModuleKeyFactory> getModuleKeyFactories() {
     return moduleKeyFactories;
   }
 
+  /**
+   * Adds a {@code ResourceReader} object that handles resource reads.
```

Comment:
My aim throughout was to make the overview comment (i.e., first sentence) useful. It worked better in some cases than in others, but I tried to be consistent. I feel that commenting the method `addFoo(Foo foo)` with "Adds a {@code Foo}" is useless because this is obvious from the method signature.